packages/blob/src/multipart/upload.ts

```diff
@@ -90,6 +88,24 @@ export function uploadPart({
     },
     options,
   );
+
+  function handleAbort(): void {
+    abortController.abort();
```

Comment:
the cancel method lives inside the `uploadAllParts` function. I moved this abortion logic into the `uploadPart` function so that we don't have to do this signal connection step on the call side (in `uploadAllParts` and in the manual methods). In the `uploadAllParts` function we `try` the `uploadPart` call and invoke `cancel` in the `catch` block here: https://github.com/vercel/storage/blob/3c7d949fff61b73ed0fba7bc82cffa0e25539d3c/packages/blob/src/multipart/upload.ts#L306

Reply:
so aborting will throw in uploadPart so that cancel is called, fine by me 👍 perhaps add a small comment at the line you linked so we understand in which case we could endup here